### PR TITLE
algorithms/afl: Disable memory limit by default

### PR DESCRIPTION
**Pull request by retrage**
## Type of PR

- [ ] Changes related to the roadmap (e.g., TODO.md) (type: A) -> Create an issue corresponding to the PR in advance, and refer to this PR in the issue.
- Changes that are not related to the roadmap
    - [ ] Change with multiple possible solutions to the issue (type: B-1) -> Create an issue corresponding to the PR in advance and refer to this PR in the issue.
    - [x] Change with a single solution (type: B-2) -> There is no need to create an issue corresponding to the PR in advance. Please discuss it in this PR.

## Related Issue

#28

## Importance of PR

- Importance of the issue
    - [ ] Large (based on several days to weeks of discussion and verification, e.g., this issue is a blocking issue for other issues on the roadmap, etc.)
    - [x] Medium (based on a few hours to a day of discussion and verification, e.g., this issue is a blocking issue for another minor issue)
    - [ ] Small (apparent changes such as build error)
- Complexity of the solution (code, tests, etc.)
    - [ ] Large (requires several days to several weeks of review)
    - [x] Medium (requires several hours to a day of review)
    - [ ] Small (trivial changes, such as build error)

## PR Overview

As already discussed in issue #28, setting the default memory limit to 50 MB causes issues in the modern environment. This PR proposes disabling this memory limit by default.

## Concerns (Optional)

- [ ] Performance
- [ ] Source Code Quality

I tested this change with several settings, but not tested with Frida mode, which affects the memory limit value. Please review in the view of Frida mode enabled @mkYYY.

---

> The PR author should fill in the following checklist when submitting this PR.

#### Optional Entries
- [ ] If this PR is a PR type A/B-1, there is a cross-link between this PR and the related issue.

#### Mandatory Entries
- [x] The PR title is a summary of the changes.
- [x] Completed each required field of the PR.

---
> The PR author should fill out the following checklist in the comments to confirm that this PR is ready to be merged

- [ ] CI is green or confirmed test run results.
- [ ] All change suggestions from reviewers have been resolved (fixed or foregone).

---
> The maintainer of this repository will set up a reviewer for each PR.
> PR reviewers should review this PR in terms of the checklist below before moving on to a detailed code review. Please comment on their initial response by filling in the checklist below.

#### Optional Entries
- [ ] The reviewer assigned more reviewers if needed.
- [ ] The reviewer noted that it is necessary to break out some of the changes in this PR into other PRs if needed.
- [ ] The reviewer noted that the initial response is insufficient if needed.

#### Mandatory Entries
- [ ] The title of this PR summarizes the changes made by this PR properly.
- [ ] The target branch of this PR is as intended.
- [ ] The reviewer understands the issues in this PR.
- [ ] The reviewer plans to review with an appropriate workload based on the importance of this PR.

---
> When the PR reviewer concludes that this PR is ready to be merged, please fill in the checklist below by posting it in the comment. If there is more than one reviewer, please do this on your own.

#### Optional Entries
- [ ] The reviewer noted that if you believe that new tests are needed to evaluate this PR, they have been noted.
- [ ] If minor refactorings are not mentioned in the PR, I understand the intent.
- [ ] If this PR is a PR type A/B-1, we have agreed on this PR's design, direction, and granularity in the related issue.

#### Mandatory Entries
- [ ] The reviewer understands how this PR addresses the issue and the specific changes.
- [ ] This PR uses the best possible issue resolution that the reviewer can think of.
- [ ] This PR is ready to be merged.

### CODE REVIEW

File: docs/algorithms/afl/algorithm_en.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ Options available are listed below:
     - `--exec_timelimit_ms=1234`
         - Specifies a time limit per PUT execution in milliseconds. The default time limit is 1 second (i.e. 1000 ms).
     - `--exec_memlimit=1234`
-        - Specifies the amount of memory available per PUT execution in megabytes. The default memory limit is 25 MB in 64-bit environment, and 50 MB in 32-bit environment.
+        - Specifies the amount of memory available per PUT execution in megabytes. The memory limit is disabled by default.
```
Comment (potetisensei) [inline docs/algorithms/afl/algorithm_en.md]:
```suggestion
        - Specifies the amount of memory available per PUT execution in megabytes. No memory limits are enforced by default.
```

File: test/algorithms/afl/loop.cpp [hunk 2]
```diff
@@ -63,7 +63,7 @@ static void AFLLoop(bool forksrv) {
   std::shared_ptr<AFLSetting> setting(new AFLSetting(
       {"../../put_binaries/libjpeg/libjpeg_turbo_fuzzer", "@@"},
       input_dir.native(), output_dir.native(), option::GetExecTimeout<AFLTag>(),
-      option::GetMemLimit<AFLTag>(), forksrv, false, /* dumb_mode*/
+      option::GetMemLimit<AFLTag>(true), forksrv, false, /* dumb_mode*/
```
Comment (potetisensei) [inline test/algorithms/afl/loop.cpp]: I think we don't need to stick to the previous value.
```suggestion
      option::GetMemLimit<AFLTag>(), forksrv, false, /* dumb_mode*/
```

**Review by potetisensei — APPROVED** [review]
No requests left from my side.

> The maintainer of this repository will set up a reviewer for each PR.
> PR reviewers should review this PR in terms of the checklist below before moving on to a detailed code review. Please comment on their initial response by filling in the checklist below.

#### Optional Entries
- [x] The reviewer assigned more reviewers if needed.
- [x] The reviewer noted that it is necessary to break out some of the changes in this PR into other PRs if needed.
- [x] The reviewer noted that the initial response is insufficient if needed.

#### Mandatory Entries
- [x] The title of this PR summarizes the changes made by this PR properly.
- [x] The target branch of this PR is as intended.
- [x] The reviewer understands the issues in this PR.
- [x] The reviewer plans to review with an appropriate workload based on the importance of this PR.

**Comment by mkYYY** [comment]
Frida mode now works too. 

---

#### Mandatory Entries
- [x] The reviewer understands how this PR addresses the issue and the specific changes.
- [x] This PR uses the best possible issue resolution that the reviewer can think of.
- [x] This PR is ready to be merged.

**Review by mkYYY — APPROVED** [review]

**Pull request merged**